web/src/main/java/org/springframework/security/web/access/RequestMatcherDelegatingAuthorizationManager.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright 2002-2020 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.security.web.access;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.function.Supplier;
+
+import javax.servlet.http.HttpServletRequest;
+
+import org.springframework.security.authorization.AuthorizationDecision;
+import org.springframework.security.authorization.AuthorizationManager;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.web.util.matcher.RequestMatcher;
+import org.springframework.security.web.util.matcher.RequestMatcherEntry;
+
+/**
+ * An {@link AuthorizationManager} which delegates to a specific
+ * {@link AuthorizationManager} based on a {@link RequestMatcher} evaluation.
+ *
+ * @author Evgeniy Cheban
+ */
+public final class RequestMatcherDelegatingAuthorizationManager implements AuthorizationManager<HttpServletRequest> {
+
+	private final List<RequestMatcherEntry<AuthorizationManager<HttpServletRequest>>> managers;
+
+	public RequestMatcherDelegatingAuthorizationManager(
+			List<RequestMatcherEntry<AuthorizationManager<HttpServletRequest>>> managers) {
+		this.managers = managers;
+	}
+
+	@Override
+	public AuthorizationDecision check(Supplier<Authentication> authentication, HttpServletRequest request) {
+		return this.managers.stream().filter((entry) -> entry.getMatcher().matches(request))
```

Comment:
Let's [avoid the use of the Stream API](https://github.com/spring-projects/spring-security/issues/7154).

Reply:
Done.